Taskfile.yml

```diff
@@ -96,6 +106,52 @@ tasks:
       - "test -e '{{.PACKAGE_VERSION_FILE}}'"
       - "test {{.TIMESTAMP | unixEpoch}} -lt $(stat --format %Y '{{.PACKAGE_VERSION_FILE}}')"
 
+  webui:
+    deps:
+      - "nodejs"
+    dir: "components/webui"
+    cmds:
+      - "rm -rf '{{.WEBUI_BUILD_DIR}}'"
+      - "mkdir -p {{.WEBUI_BUILD_DIR}}"
+      - "meteor npm install --production"
+      - "meteor build --directory {{.WEBUI_BUILD_DIR}}"
+      - >-
+        rsync -a
+        "{{.WEBUI_BUILD_DIR}}/bundle/"
+        "$PWD/settings.json"
+        "$PWD/launcher.js"
+        "{{.WEBUI_BUILD_DIR}}/"
+      - "rm -rf '{{.WEBUI_BUILD_DIR}}/bundle/'"
+      - |-
+        cd {{.WEBUI_BUILD_DIR}}/programs/server
+        PATH={{.NODEJS_BIN_DIR}}:$PATH $(readlink -f {{.NODEJS_BIN_DIR}}/npm) install
+    sources:
+      - "./.meteor/*"
+      - "./client/**/*"
+      - "./imports/**/*"
+      - "./server/**/*"
+      - "./tests/**/*"
+      - "./*"
+
+  nodejs:
+    vars:
+      NODEJS_VERSION: "14.21.3"
+      NODEJS_RELEASE: "v{{.NODEJS_VERSION}}-linux-x64"
+      TAR_FILENAME: "node-{{.NODEJS_RELEASE}}.tar.xz"
+      TAR_FILE_PATH: "{{.NODEJS_BUILD_DIR}}/{{.TAR_FILENAME}}"
+    cmds:
+      - "rm -rf '{{.NODEJS_BUILD_DIR}}/node'"
+      - "mkdir -p {{.NODEJS_BUILD_DIR}}"
+      - >-
+        curl -fsSL
+        https://nodejs.org/dist/v{{.NODEJS_VERSION}}/{{.TAR_FILENAME}}
+        -o {{.TAR_FILE_PATH}}
+      - "tar xf {{.TAR_FILE_PATH}} -C {{.NODEJS_BUILD_DIR}}"
+      - "mv {{.NODEJS_BUILD_DIR}}/node-{{.NODEJS_RELEASE}} {{.NODEJS_BUILD_DIR}}/node"
+      - "rm -f '{{.TAR_FILE_PATH}}'"
+    sources:
```

Comment:
If I'm not mistaken, you'll need to use `status` like we are for other components. Otherwise, if a component is deleted manually, `task` won't detect it and regenerate it.

Reply:
How currently this works is:

(A: empty NODEJS_BUILD_DIR/node) ->
`task nodejs` ->
(B: valid NODEJS_BUILD_DIR/node) ->
`task nodejs` which finds `A != B` and actually rebuilds the conf -> 
(C: valid NODEJS_BUILD_DIR/node) ->
`task nodejs` which finds `B == C` and avoid rebuilding the conf.

As discussed offline, the approach should be applied to the `webui` conf above.